### PR TITLE
Enables Charmhub publish job

### PR DESCRIPTION
**Pull request by DnPlas**
This PR updates publish job and all necessary changes to make it possible to test and push the bundle to Charmhub:
* bundle*.yaml: updates applications repo charmstore -> charmhub
* workflows/publish.yaml: updates publish job with charmhub-upload-action
* tests/test_kubectl.py: enables training-operator checks and removes pytorch/tf

### CODE REVIEW

File: bundle.yaml [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
-  dex-auth:                  { charm: cs:dex-auth-60,                 scale: 1 }
-  istio-ingressgateway:      { charm: cs:istio-ingressgateway-20,     scale: 1 }
-  istio-pilot:               { charm: cs:istio-pilot-20,              scale: 1, options: { default-gateway: "kubeflow-gateway" } }
-  jupyter-controller:        { charm: cs:jupyter-controller-56,       scale: 1 }
-  jupyter-ui:                { charm: cs:jupyter-ui-10,               scale: 1 }
-  katib-controller:          { charm: cs:katib-controller-30,         scale: 1 }
+  admission-webhook:         { charm: admission-webhook,       channel: edge, scale: 1 }
+  argo-controller:           { charm: argo-controller,         channel: edge, scale: 1 }
+  dex-auth:                  { charm: dex-auth,                channel: edge, scale: 1 }
+  istio-ingressgateway:      { charm: istio-ingressgateway,    channel: edge, scale: 1 }
```
Comment (DomFleischmann) [inline bundle.yaml:7]: This should be pointing to `istio-gateway` with `kind: ingress`

Comment (DomFleischmann) [reply bundle.yaml:7]: You can check the latest changes in notebook-operator to see an example of how the bundle should look

File: bundle.yaml [hunk 2]
```diff
@@ -1,33 +1,33 @@
...
-  oidc-gatekeeper:           { charm: cs:oidc-gatekeeper-54,          scale: 1 }
-  pytorch-operator:          { charm: cs:pytorch-operator-53,         scale: 1 }
+  kfp-persistence:           { charm: kfp-persistence,         channel: edge, scale: 1 }
+  kfp-schedwf:               { charm: kfp-schedwf,             channel: edge, scale: 1 }
+  kfp-ui:                    { charm: kfp-ui,                  channel: edge, scale: 1 }
+  kfp-viewer:                { charm: kfp-viewer,              channel: edge, scale: 1 }
+  kfp-viz:                   { charm: kfp-viz,                 channel: edge, scale: 1 }
+  kubeflow-dashboard:        { charm: kubeflow-dashboard,      channel: edge, scale: 1 }
+  kubeflow-profiles:         { charm: kubeflow-profiles,       channel: edge, scale: 1 }
+  kubeflow-volumes:          { charm: kubeflow-volumes,        channel: edge, scale: 1 }
```
Comment (DomFleischmann) [inline bundle.yaml:24]: Please also add the kubeflow-tensorboard charms which should relate with istio-pilot

File: bundle.yaml [hunk 3]
```diff
@@ -1,33 +1,33 @@
...
+  mlmd:                      { charm: mlmd,                    channel: edge, scale: 1 }
+  minio:                     { charm: minio,                   channel: edge, scale: 1 }
```
Comment (DomFleischmann) [inline bundle.yaml:26]: Also add the metacontroller charm

File: bundle.yaml [hunk 4]
```diff
@@ -1,33 +1,33 @@
...
-  minio:                     { charm: cs:minio-55,                    scale: 1 }
-  oidc-gatekeeper:           { charm: cs:oidc-gatekeeper-54,          scale: 1 }
-  pytorch-operator:          { charm: cs:pytorch-operator-53,         scale: 1 }
+  kfp-persistence:           { charm: kfp-persistence,         channel: edge, scale: 1 }
+  kfp-schedwf:               { charm: kfp-schedwf,             channel: edge, scale: 1 }
+  kfp-ui:                    { charm: kfp-ui,                  channel: edge, scale: 1 }
+  kfp-viewer:                { charm: kfp-viewer,              channel: edge, scale: 1 }
+  kfp-viz:                   { charm: kfp-viz,                 channel: edge, scale: 1 }
+  kubeflow-dashboard:        { charm: kubeflow-dashboard,      channel: edge, scale: 1 }
+  kubeflow-profiles:         { charm: kubeflow-profiles,       channel: edge, scale: 1 }
```
Comment (DomFleischmann) [inline bundle.yaml:23]: Please add the kubeflow-roles charm

File: bundle.yaml [hunk 5]
```diff
@@ -1,33 +1,33 @@
...
+  kubeflow-volumes:          { charm: kubeflow-volumes,        channel: edge, scale: 1 }
+  mlmd:                      { charm: mlmd,                    channel: edge, scale: 1 }
```
Comment (DomFleischmann) [inline bundle.yaml:25]: Please add envoy which relates with mlmd.

File: bundle.yaml [hunk 6]
```diff
@@ -48,3 +57,4 @@ relations:
 - [kfp-api:object-storage, minio:object-storage]
 - [kfp-ui:object-storage, minio:object-storage]
 - [kubeflow-profiles, kubeflow-dashboard]
+- [mlmd:grpc, envoy:grpc]
```
Comment (DomFleischmann) [inline bundle.yaml:60]: this is misssing a releation between kfp-profile-controller and minio

File: bundle.yaml [hunk 7]
```diff
@@ -1,33 +1,41 @@
...
+  kfp-persistence:               { charm: kfp-persistence,         channel: latest/edge, scale: 1 }
+  kfp-profile-controller:        { charm: kfp-profile-controller,  channel: latest/edge, scale: 1 }
+  kfp-schedwf:                   { charm: kfp-schedwf,             channel: latest/edge, scale: 1 }
+  kfp-ui:                        { charm: kfp-ui,                  channel: latest/edge, scale: 1 }
+  kfp-viewer:                    { charm: kfp-viewer,              channel: latest/edge, scale: 1 }
+  kfp-viz:                       { charm: kfp-viz,                 channel: latest/edge, scale: 1 }
+  kubeflow-dashboard:            { charm: kubeflow-dashboard,      channel: latest/edge, scale: 1 }
+  kubeflow-profiles:             { charm: kubeflow-profiles,       channel: latest/edge, scale: 1 }
+  kubeflow-volumes:              { charm: kubeflow-volumes,        channel: latest/edge, scale: 1 }
+  metacontroller-operator:       { charm: metacontroller-operator, channel: latest/edge, scale: 1 }
```
Comment (ca-scribner) [inline bundle.yaml:28]: Fixes "metacontroller needs trust" error
```suggestion
  metacontroller-operator:       { charm: metacontroller-operator, channel: latest/edge, scale: 1, trust: true }
```

File: bundle.yaml (hunk 1 above)
Comment (DnPlas) [reply bundle.yaml]: Done

File: bundle.yaml (hunk 4 above)
Comment (DnPlas) [reply bundle.yaml]: I left that charm commented out as it is not ready.

File: bundle-lite.yaml [hunk 8]
```diff
@@ -1,28 +1,29 @@
...
-  tfjob-operator:            { charm: cs:tfjob-operator-1,            scale: 1 }
+  admission-webhook:             { charm: admission-webhook,       channel: edge, scale: 1 }
+  argo-controller:               { charm: argo-controller,         channel: edge, scale: 1 }
+  dex-auth:                      { charm: dex-auth,                channel: edge, scale: 1 }
+  istio-ingressgateway-operator: { charm: istio-gateway,           channel: latest/edge, scale: 1, trust: true, options: { kind: ingress } }
+  istio-pilot:                   { charm: istio-pilot,             channel: edge, scale: 1, options: { default-gateways: "kubeflow-gateway" } }
+  jupyter-controller:            { charm: jupyter-controller,      channel: edge, scale: 1 }
+  jupyter-ui:                    { charm: jupyter-ui,              channel: edge, scale: 1 }
+  kfp-api:                       { charm: kfp-api,                 channel: edge, scale: 1 }
+  kfp-db:                        { charm: cs:~charmed-osm/mariadb-k8s-35, scale: 1, options: { database: mlpipeline } }
```
Comment (DomFleischmann) [inline bundle-lite.yaml:12]: We should point to this chamr in [Charmhub](https://charmhub.io/charmed-osm-mariadb-k8s)

File: bundle.yaml [hunk 9]
```diff
@@ -1,33 +1,41 @@
...
-  minio:                     { charm: cs:minio-55,                    scale: 1 }
-  oidc-gatekeeper:           { charm: cs:oidc-gatekeeper-54,          scale: 1 }
-  pytorch-operator:          { charm: cs:pytorch-operator-53,         scale: 1 }
-  spark:                     { charm: cs:~spark-charmers/spark-2,     scale: 1 }
-  seldon-controller-manager: { charm: cs:seldon-core-50,              scale: 1 }
-  tfjob-operator:            { charm: cs:tfjob-operator-1,            scale: 1 }
+  admission-webhook:             { charm: admission-webhook,       channel: latest/edge, scale: 1 }
+  argo-controller:               { charm: argo-controller,         channel: latest/edge, scale: 1 }
+  argo-server:                   { charm: argo-server,             channel: latest/edge, scale: 1 }
+  dex-auth:                      { charm: dex-auth,                channel: latest/edge, scale: 1 }
```
Comment (ca-scribner) [inline bundle.yaml:7]: dex-auth needs `--trust`
```suggestion
  dex-auth:                      { charm: dex-auth,                channel: latest/edge, scale: 1, trust: true }
```

File: bundle.yaml [hunk 10]
```diff
@@ -1,33 +1,40 @@
...
+  kfp-db:                        { charm: charmed-osm-mariadb-k8s, channel: latest/edge, scale: 1, options: { database: mlpipeline } }
+  kfp-persistence:               { charm: kfp-persistence,         channel: latest/edge, scale: 1 }
+  kfp-profile-controller:        { charm: kfp-profile-controller,  channel: latest/edge, scale: 1 }
+  kfp-schedwf:                   { charm: kfp-schedwf,             channel: latest/edge, scale: 1 }
+  kfp-ui:                        { charm: kfp-ui,                  channel: latest/edge, scale: 1 }
+  kfp-viewer:                    { charm: kfp-viewer,              channel: latest/edge, scale: 1 }
+  kfp-viz:                       { charm: kfp-viz,                 channel: latest/edge, scale: 1 }
+  kubeflow-dashboard:            { charm: kubeflow-dashboard,      channel: latest/edge, scale: 1 }
+  kubeflow-profiles:             { charm: kubeflow-profiles,       channel: latest/edge, scale: 1 }
+  kubeflow-volumes:              { charm: kubeflow-volumes,        channel: latest/edge, scale: 1 }
```
Comment (DomFleischmann) [inline bundle.yaml:27]: The `kubeflow-tensorboards` charm is still missing here.

**Comment by DnPlas** [comment]
Closing this, tracking on #421

**Pull request closed without merging**